cparser/processor.js

```diff
@@ -41,8 +41,16 @@ module.exports = that = {
 				return false;
 			}
 
-			if ((fileBuffer.indexOf('#pragma SPARK_NO_PREPROCESSOR') >= 0) ||
-				(fileBuffer.indexOf('#pragma PARTICLE_NO_PREPROCESSOR') >= 0) ||
+			var noPreprocessorMatch = fileBuffer.match('#pragma (SPARK_NO_PREPROCESSOR|PARTICLE_NO_PREPROCESSOR)');
+			if (noPreprocessorMatch) {
```

Comment:
Currently `#pragma PARTICLE_NO_PREPROCESSOR` does NOT seem to disable the include of application.h, and I can't see why it doesn't from the code replaced.